theano/gpuarray/opt.py

```diff
@@ -656,6 +669,8 @@ def local_gpu_contiguous_gpu_contiguous(node):
     if isinstance(node.op, GpuContiguous):
         inp = node.inputs[0]
         if inp.owner and isinstance(inp.owner.op, GpuContiguous):
+            if not getattr(inp.tag, 'trace', None):
+                copy_stack_trace(node.outputs[0], inp)
```

Comment:
Hmm. Maybe I don't need to copy the trace at all here, since the optimization doesn't add any nodes.